src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,142 @@
-import { NewCommentForm } from '../NewCommentForm';
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+} from 'react';
+import {
+  getPostComments,
+  Comment,
+  createComment,
+  deleteComment,
+} from '../../api/comments';
+import { getPostDetails } from '../../api/post';
+import { NewCommentForm, NewComment } from '../NewCommentForm';
+import { Post } from '../PostsList';
+import { Loader } from '../Loader';
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type Props = {
+  postId: number,
+};
+
+export const PostDetails: React.FC<Props> = ({ postId }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [post, setPost] = useState<Post>();
+
+  const [showComments, setShowComments] = useState(true);
+  const [loadingPost, setLoadingPost] = useState(false);
+  const [loadingComments, setLoadingComments] = useState(false);
+  const [idDeletedComment, setIdDeletedComment] = useState(0);
+  const [newComment, setNewComment] = useState<NewComment>();
+
+  useEffect(() => {
+    getPostDetails(postId)
+      .then(receivedPost => {
+        setPost(receivedPost);
+        setLoadingPost(true);
+      });
+  }, [postId]);
+
+  const getComments = useCallback(() => {
+    getPostComments(postId)
+      .then(receivedComments => {
+        setComments(receivedComments);
+        setLoadingComments(true);
+      });
+  }, [postId, idDeletedComment, newComment]);
+
+  useEffect(getComments, [postId, idDeletedComment, newComment]);
+
+  useMemo(async () => {
+    if (idDeletedComment) {
+      await deleteComment(idDeletedComment);
+    }
+
+    getComments();
+  }, [idDeletedComment]);
+
+  useMemo(async () => {
+    if (newComment) {
+      await createComment({
+        ...newComment,
+        id: comments.length,
+        postId,
+        createdAt: '',
+        updatedAt: '',
+      });
+    }
+
+    getComments();
+  }, [newComment]);
```

Comment:
I think you didn't understand the usage of useMemo, remove it for these code and just wrap them in standard function, then use it where you need it

https://dmitripavlutin.com/react-usememo-hook/